### PR TITLE
[REG-1338] Sanitize action name strings

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Sanitizes all special characters out of action name strings and replaces with `_`
  - I played with trimming out the characters instead of replacing them, but that led to more frequent name conflicts in testing
- Input
```csharp
[RGState("1Some-/=+Sta7e 3 World")]
public bool someState;

[RGAction("1Perform-/=+At7ack 3 Now")]
public void PerformAttack(bool isEnabled, string actionName, int attackNumber, float damageAmount)
{
    
}
```
- Output [RGAction]
```csharp
/*
* This file has been automatically generated. Do not modify.
*/

using System;
using System.Collections.Generic;
using RegressionGames;
using RegressionGames.RGBotConfigs;
using RegressionGames.StateActionTypes;
using UnityEngine;
using Unity.BossRoom.Gameplay.Actions;

namespace RegressionGames
{
    public class RGAction_1Perform____At7ack_3_Now : RGAction
    {
        public void Start()
        {
            AddMethod("1Perform-/=+At7ack 3 Now", new Action<bool, string, int, float>(GetComponent<MeleeAction>().PerformAttack));
        }

        public override string GetActionName()
        {
            return "1Perform-/=+At7ack 3 Now";
        }

        public override void StartAction(Dictionary<string, object> input)
        {
            if (!input.ContainsKey("isEnabled"))
            {
                RGDebug.LogError("No parameter 'isEnabled' found");
                return;
            }

            string isEnabledInput = input["isEnabled"].ToString();
            bool isEnabled = default;
            try
            {
                bool.TryParse(isEnabledInput, out isEnabled);
            }
            catch (Exception ex)
            {
                RGDebug.LogError("Failed to parse 'isEnabled'");
                RGDebug.LogError(ex.Message);
            }
            finally
            {
            }

            if (!input.ContainsKey("actionName"))
            {
                RGDebug.LogError("No parameter 'actionName' found");
                return;
            }

            string actionNameInput = input["actionName"].ToString();
            string actionName = default;
            try
            {
                actionName = actionNameInput;
            }
            catch (Exception ex)
            {
                RGDebug.LogError("Failed to parse 'actionName'");
                RGDebug.LogError(ex.Message);
            }
            finally
            {
            }

            if (!input.ContainsKey("attackNumber"))
            {
                RGDebug.LogError("No parameter 'attackNumber' found");
                return;
            }

            string attackNumberInput = input["attackNumber"].ToString();
            int attackNumber = default;
            try
            {
                int.TryParse(attackNumberInput, out attackNumber);
            }
            catch (Exception ex)
            {
                RGDebug.LogError("Failed to parse 'attackNumber'");
                RGDebug.LogError(ex.Message);
            }
            finally
            {
            }

            if (!input.ContainsKey("damageAmount"))
            {
                RGDebug.LogError("No parameter 'damageAmount' found");
                return;
            }

            string damageAmountInput = input["damageAmount"].ToString();
            float damageAmount = default;
            try
            {
                float.TryParse(damageAmountInput, out damageAmount);
            }
            catch (Exception ex)
            {
                RGDebug.LogError("Failed to parse 'damageAmount'");
                RGDebug.LogError(ex.Message);
            }
            finally
            {
            }

            Invoke("1Perform-/=+At7ack 3 Now", isEnabled, actionName, attackNumber, damageAmount);
        }
    }

    public class RGActionRequest_1Perform____At7ack_3_Now : RGActionRequest
    {
        public RGActionRequest_1Perform____At7ack_3_Now(bool isEnabled, string actionName, int attackNumber, float damageAmount)
        {
            action = "1Perform-/=+At7ack 3 Now";
            Input = new ()
            {{"isEnabled", isEnabled}, {"actionName", actionName}, {"attackNumber", attackNumber}, {"damageAmount", damageAmount}, };
        }
    }
}
```
- Output [RGState]
```csharp
/*
* This file has been automatically generated. Do not modify.
*/

using System;
using System.Collections.Generic;
using Unity.BossRoom.Gameplay.Actions;

namespace RegressionGames.RGBotConfigs
{
    public class MeleeAction_RGState : RGState
    {
        private MeleeAction myComponent;
        public void Start()
        {
            myComponent = this.GetComponent<MeleeAction>();
            if (myComponent == null)
            {
                RGDebug.LogError("MeleeAction not found");
            }
        }

        public override Dictionary<string, object> GetState()
        {
            var state = new Dictionary<string, object>();
            state.Add("1Some-/=+Sta7e 3 World", myComponent.someState);
            return state;
        }
    }
}
```

### CODE REVIEW

File: Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using System.Text.RegularExpressions;
+
+namespace RegressionGames
+{
+    public class CodeGeneratorUtils
+    {
+        public static string SanitizeActionName(string name)
+        {
+            return Regex.Replace(name.Replace(" ", "_"), "[^0-9a-zA-Z_]", "_");
```
Comment (vontell) [inline Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs:9]: Good catch! Not sure if we need this now, but we may still get invalid names, like an action or state that is entirely numbers or starts with a number.

File: Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System.Text.RegularExpressions;
+
+namespace RegressionGames
+{
+    public class CodeGeneratorUtils
+    {
+        public static string SanitizeActionName(string name)
```
Comment (vontell) [inline Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs:7]: The same code should also apply to states, I believe? Can we also fix that here, or we can wait until we see where we get with RGState

**Review by vontell — CHANGES_REQUESTED** [review]

**Review by addisonbgross — APPROVED** [review]

File: Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs (hunk 2 above)
Comment (vontell) [reply Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs:7]: Oh interesting, thanks!

File: Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs (hunk 1 above)
Comment (vontell) [reply Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs:9]: Ah ok I see, thanks!

**Review by vontell — APPROVED** [review]

**Pull request merged**